sdk/java/pulumi/src/main/java/com/pulumi/test/PulumiTest.java

```diff
@@ -10,12 +14,127 @@
  */
 public interface PulumiTest {
 
+    /**
+     * Run a Pulumi test stack callback, wait for result and return {@link TestResult}.
+     *
+     * @param stack the stack to run in Pulumi test runtime
+     * @return a {@link TestResult} from Pulumi test runtime after running the stack
+     * @see #runTestAsync(Consumer)
+     * @see #withOptions(TestOptions)
+     * @see #withMocks(Mocks)
+     */
+    static TestResult runTest(Consumer<Context> stack) {
```

Comment:
`PulumiTest` echoes the `Pulumi` interface semantics, with the static methods for DSL-style of API.